### PR TITLE
Add Profile Page

### PR DESCRIPTION
**Pull request by maruthimohan**
Hey @DavidGolodetsky , this PR is for the task #41 

**Changes as a part of this PR:**

- Added a new route to navigate to the Profile page
- Added a new component to view the current user's username and update that
- Added a new PUT method to update the current user's username in MongoDB

### CODE REVIEW

File: client/src/components/UserProfile.vue [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+						color="secondary"
+						:disabled="!formIsValid"
+						@click="updateUserData"
+					>
+						{{ isReadOnly ? "Edit" : "Save" }}
+					</v-btn>
+				</v-card-actions>
+			</v-col>
+		</v-row>
+		<v-snackbar
```
Comment (DavidGolodetsky) [inline client/src/components/UserProfile.vue:27]: There is TheAlert component that could be used as a notification

**Review by DavidGolodetsky — COMMENTED** [review]
Thank you! I've add some comment, they're not required to do but if you want you can take a look at them. Anyway I'll merge it:)

File: client/src/views/Profile.vue [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<template>
+	<section>
+		<div class="d-flex my-3">
```
[the remark this answers is not in the document]
Comment (maruthimohan) [reply client/src/views/Profile.vue]: Hey @DavidGolodetsky , replaced the code with `TheTitle` component for the title.

File: client/src/components/UserProfile.vue [hunk 3]
```diff
@@ -0,0 +1,123 @@
+<template>
+	<v-container>
```
[the remark this answers is not in the document]
Comment (maruthimohan) [reply client/src/components/UserProfile.vue]: I agree, removed the extra component, and placed all the code in the `Profile` view.

File: client/src/components/UserProfile.vue (hunk 1 above)
Comment (maruthimohan) [reply client/src/components/UserProfile.vue]: Hey @DavidGolodetsky , makes sense, replaced `VSnackbar` with `TheAlert` component.

**Comment by maruthimohan** [comment]
Hey @DavidGolodetsky , made a couple of changes as per your review.

**Comment by DavidGolodetsky** [comment]
Thank you!

**Pull request merged**